content/blog/analysis-with-RCrunch.Rmd

```diff
@@ -0,0 +1,300 @@
+```{r}
+categories(so_data$HomeRemote)
+```
+
+One way we could try to fix this would be to download the column again, substitute all NAs with `NA`, reorder the factor, and then send them back to Crunch.
+```{r}
+clearCache() # let's make this a fair fight
+system.time({
+    home_var <- as.vector(so_data$HomeRemote)
+    home_var[home_var == "NA"] <- NA
+    home_var <- droplevels(home_var)
+    home_var <- factor(home_var, levels = c("Never",
+                                            "A few days each month",
+                                            "Less than half the time, but at least one day each week",
+                                            "About half the time",
+                                            "More than half, but not all, the time",
+                                            "All or almost all the time (I'm full-time remote)",
+                                            "It's complicated"))
+    so_data[['HomeRemoteCleaned']] <- home_var
+})
+```
+
+But we can do this even quicker and easier with the power of Crunch. Not only are these changes fewer lines of R code, but they also make their changes on the server, so any collaborators will be able to see the changes you make immediately. No more waiting around to download data or send Crunch a new round of cleaned exports. 
+```{r}
+clearCache() # let's make this a fair fight
+system.time({
+    is.na(categories(so_data$HomeRemote)) <- "NA"
+    indeces <- match(c("Never",
+                       "A few days each month",
+                       "Less than half the time, but at least one day each week",
+                       "About half the time",
+                       "More than half, but not all, the time",
+                       "All or almost all the time (I'm full-time remote)",
+                       "It's complicated", "NA", "No Data"), names(categories(so_data$HomeRemote)))
+    categories(so_data$HomeRemote) <- categories(so_data$HomeRemote)[indeces]    
+})
+```
+
+Now the categories look much better: "NA" is marked as missing, the responses are ordered from least to most home/remote time.
+```{r}
+categories(so_data$HomeRemote)
+```
+
+Categorical variables in Crunch support more complex missing value than R does, multiple categories can be marked as missing: there's not just one single "NA" value. For this variable, the "It's complicated" response doesn't really fit in with the other responses either. With Crunch we can mark it as missing so that it doesn't show up like other categories, but importantly, all of that data is still there if we need it. Additionally, we can set numeric values for each level.
+```{r}
+is.na(categories(so_data$HomeRemote)) <- "It's complicated"
+values(categories(so_data$HomeRemote)) <- c(0, 1, 2, 3, 4, 5, NA, NA, NA)
```

Comment:
Did you consider some other values like “rough percentage of remoteness”? like
`0, 5, 25, 50, 75, 100` ? You could say potentially interesting things about the numeric values that way. / just another way to look at the data